aws-iot-mitigationaction/src/test/java/com/amazonaws/iot/mitigationaction/CreateHandlerTest.java

```diff
@@ -0,0 +1,173 @@
+                .roleArn(MITIGATION_ACTION_ROLE_ARN)
+                .actionParams(ACTION_PARAMS_WITH_REPLACE_DEFAULT_POLICY_VERSION_PARAMS)
+                .tags(MODEL_TAGS)
+                .build();
+
+        ResourceHandlerRequest<ResourceModel> request = ResourceHandlerRequest.<ResourceModel>builder()
+                .desiredResourceState(model)
+                .logicalResourceIdentifier("TestMitigationActionIdentifier")
+                .clientRequestToken(CLIENT_REQUEST_TOKEN)
+                .desiredResourceTags(DESIRED_TAGS)
+                .build();
+
+        when(proxy.injectCredentialsAndInvokeV2(any(), any()))
+                .thenThrow(ResourceAlreadyExistsException.builder().build());
+
+        ProgressEvent<ResourceModel, CallbackContext> progressEvent =
+                handler.handleRequest(proxy, request, null, logger);
+        assertThat(progressEvent.getErrorCode()).isEqualTo(HandlerErrorCode.AlreadyExists);
+    }
+
+    @Test
+    public void handleRequest_NoName_GeneratedByHandler() {
+
+        ResourceModel model = ResourceModel.builder()
+                .roleArn(MITIGATION_ACTION_ROLE_ARN)
+                .actionParams(ACTION_PARAMS_WITH_REPLACE_DEFAULT_POLICY_VERSION_PARAMS)
+                .tags(MODEL_TAGS)
+                .build();
+
+        ResourceHandlerRequest<ResourceModel> request = ResourceHandlerRequest.<ResourceModel>builder()
+                .desiredResourceState(model)
+                .logicalResourceIdentifier("MyResourceName")
+                .clientRequestToken("MyToken")
+                .desiredResourceTags(DESIRED_TAGS)
+                .stackId("arn:aws:cloudformation:us-east-1:123456789012:stack/my-stack-name/" +
+                        "084c0bd1-082b-11eb-afdc-0a2fadfa68a5")
+                .build();
+
+        CreateMitigationActionResponse createApiResponse = CreateMitigationActionResponse.builder()
+                .actionId(ACTION_ID)
+                .actionArn(ACTION_ARN)
+                .build();
+        when(proxy.injectCredentialsAndInvokeV2(any(), any()))
+                .thenReturn(createApiResponse);
+
+        handler.handleRequest(proxy, request, null, logger);
+
+        ArgumentCaptor<CreateMitigationActionRequest> requestCaptor =
+                ArgumentCaptor.forClass(CreateMitigationActionRequest.class);
+        verify(proxy).injectCredentialsAndInvokeV2(requestCaptor.capture(), any());
+
+        CreateMitigationActionRequest submittedRequest = requestCaptor.getValue();
+
+        // Can't easily check the randomly generated name. Just making sure it contains part of
+        // the logical identifier and the stack name, and some more random characters
+        assertThat(submittedRequest.actionName()).contains("my-stack");
+        assertThat(submittedRequest.actionName()).contains("MyRes");
+        assertThat(submittedRequest.actionName().length() > 20).isTrue();
+    }
+
+    // TODO: test system tags when the src code is ready
```

Comment:
Remove this line I'd say. When we add support then it's possible we don't even look at this line and just forget about it. Hardly people check for TODOs in code.

Reply:
Removed